src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -330,6 +330,82 @@ private static async Task<BinaryData> FromStreamAsync(Stream stream, bool async,
             }
         }
 
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance from the specified file.
+        /// </summary>
+        /// <param name="path">The path to the file.</param>
+        /// <returns>A value representing all of the data from the file.</returns>
+        public static BinaryData FromFile(string path)
+        {
+            if (path is null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            return FromFileAsync(path, async: false).GetAwaiter().GetResult();
+        }
```

Comment:
```suggestion
        public static BinaryData FromFile(string path) => FromFile(path, mediaType: null);
```
? Same for other such cases?

Reply:
Changed for all cases